<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Stegy</class>
 <widget class="QMainWindow" name="Stegy">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>944</width>
    <height>344</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Stegy</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="icon">
        <iconset>
         <normaloff>images/embed.png</normaloff>images/embed.png</iconset>
       </attribute>
       <attribute name="title">
        <string>Embedding</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="QGroupBox" name="groupBox_6">
          <property name="title">
           <string>Preview</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_7">
           <item>
            <widget class="QLabel" name="cover_label">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximumSize">
              <size>
               <width>205</width>
               <height>205</height>
              </size>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="pixmap">
              <pixmap>images/bg.jpg</pixmap>
             </property>
             <property name="scaledContents">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="embed_pushButton">
             <property name="text">
              <string>Embed</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QGroupBox" name="groupBox">
              <property name="title">
               <string>Files</string>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_4">
               <item>
                <layout class="QGridLayout" name="gridLayout">
                 <item row="0" column="0">
                  <widget class="QLabel" name="label">
                   <property name="text">
                    <string>Cover File</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="2">
                  <widget class="QPushButton" name="coverFileButton">
                   <property name="text">
                    <string>Get Cover File</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLabel" name="label_2">
                   <property name="text">
                    <string>Embed File</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="2">
                  <widget class="QPushButton" name="EmbedFileButton">
                   <property name="text">
                    <string>Get Embed File</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="groupBox_3">
              <property name="title">
               <string>Compression</string>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <item>
                <layout class="QVBoxLayout" name="verticalLayout">
                 <item>
                  <widget class="QCheckBox" name="compress_checkBox">
                   <property name="text">
                    <string>Compress</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_3">
                   <property name="text">
                    <string>Compression Level (1-9)</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="complevelspinBox">
                   <property name="enabled">
                    <bool>false</bool>
                   </property>
                   <property name="minimum">
                    <number>1</number>
                   </property>
                   <property name="maximum">
                    <number>9</number>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QGroupBox" name="groupBox_2">
              <property name="title">
               <string>Encryption</string>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <item>
                <layout class="QGridLayout" name="gridLayout_2">
                 <item row="0" column="0">
                  <widget class="QLabel" name="label_4">
                   <property name="text">
                    <string>Type</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="QComboBox" name="algo_comboBox"/>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLabel" name="label_5">
                   <property name="text">
                    <string>Mode</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="0">
                  <widget class="QLabel" name="label_6">
                   <property name="text">
                    <string>Key</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
                  <widget class="QLineEdit" name="pass_lineEdit"/>
                 </item>
                 <item row="1" column="1">
                  <widget class="QComboBox" name="mode_comboBox"/>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="groupBox_4">
              <property name="title">
               <string>Other Options</string>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_3">
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_2">
                 <item>
                  <widget class="QCheckBox" name="nocrc32_checkBox">
                   <property name="text">
                    <string>No CRC32 Checksum</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QCheckBox" name="efn_checkBox">
                   <property name="text">
                    <string>Embed FileName</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="icon">
        <iconset>
         <normaloff>images/extract.png</normaloff>images/extract.png</iconset>
       </attribute>
       <attribute name="title">
        <string>Extraction</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QGroupBox" name="groupBox_5">
          <property name="maximumSize">
           <size>
            <width>789</width>
            <height>290</height>
           </size>
          </property>
          <property name="title">
           <string>Files</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_9">
           <item>
            <layout class="QGridLayout" name="gridLayout_5">
             <property name="sizeConstraint">
              <enum>QLayout::SetDefaultConstraint</enum>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="label_7">
               <property name="text">
                <string>Stego File</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="est_lineEdit"/>
             </item>
             <item row="0" column="2">
              <widget class="QPushButton" name="stegoFileButton">
               <property name="text">
                <string>Open</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_9">
               <property name="text">
                <string>Output File</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="of_lineEdit">
               <property name="enabled">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QCheckBox" name="uon_checkBox">
               <property name="text">
                <string>Use Original Name</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item row="2" column="1" colspan="2">
              <widget class="QLineEdit" name="key_lineEdit"/>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_8">
               <property name="text">
                <string>Key</string>
               </property>
              </widget>
             </item>
             <item row="4" column="2">
              <widget class="QPushButton" name="extract_pushButton">
               <property name="text">
                <string>Extract</string>
               </property>
              </widget>
             </item>
             <item row="4" column="0" colspan="2">
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="3" column="0" colspan="3">
              <spacer name="verticalSpacer">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="icon">
        <iconset>
         <normaloff>images/about.png</normaloff>images/about.png</iconset>
       </attribute>
       <attribute name="title">
        <string>About</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QTextEdit" name="textEdit">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="html">
           <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Ubuntu'; font-size:11pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Stegy &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;This application is a frontend to the command line tool steghide. This application is written using pure Ruby using Qt Framework. &lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Programming: Sibi&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="textInteractionFlags">
           <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>compress_checkBox</sender>
   <signal>clicked(bool)</signal>
   <receiver>complevelspinBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>411</x>
     <y>231</y>
    </hint>
    <hint type="destinationlabel">
     <x>431</x>
     <y>289</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>coverFileButton</sender>
   <signal>clicked()</signal>
   <receiver>Stegy</receiver>
   <slot>getCoverFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>404</x>
     <y>97</y>
    </hint>
    <hint type="destinationlabel">
     <x>463</x>
     <y>-6</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>EmbedFileButton</sender>
   <signal>clicked()</signal>
   <receiver>Stegy</receiver>
   <slot>getEmbedFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>461</x>
     <y>147</y>
    </hint>
    <hint type="destinationlabel">
     <x>622</x>
     <y>4</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>algo_comboBox</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>Stegy</receiver>
   <slot>changeModeEntries()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>680</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>748</x>
     <y>-2</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stegoFileButton</sender>
   <signal>clicked()</signal>
   <receiver>Stegy</receiver>
   <slot>getStegoFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>775</x>
     <y>88</y>
    </hint>
    <hint type="destinationlabel">
     <x>743</x>
     <y>-8</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>embed_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>Stegy</receiver>
   <slot>embed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>125</x>
     <y>306</y>
    </hint>
    <hint type="destinationlabel">
     <x>326</x>
     <y>3</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uon_checkBox</sender>
   <signal>clicked(bool)</signal>
   <receiver>of_lineEdit</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>712</x>
     <y>120</y>
    </hint>
    <hint type="destinationlabel">
     <x>676</x>
     <y>121</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>extract_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>Stegy</receiver>
   <slot>extract()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>754</x>
     <y>297</y>
    </hint>
    <hint type="destinationlabel">
     <x>602</x>
     <y>-4</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>getCoverFile()</slot>
  <slot>getEmbedFile()</slot>
  <slot>changeModeEntries()</slot>
  <slot>getStegoFile()</slot>
  <slot>embed()</slot>
  <slot>extract()</slot>
 </slots>
</ui>
